Activities/Module13 Activity.md

```diff
@@ -146,3 +146,217 @@ In case your server is is not responding to requests after a long period of inac
+     * Change the "start" script in your server/package.json from "ts-node server.ts" to "node dist/server.js".
+     * Add a new script called "start:dev" with the value "ts-node server.ts". You can use this for your local development.
+   * The package.json file should look something like this:
+     * [`package.json`](Module13Assets/package.json) - Contains required dependencies and scripts
+
+   * You need to make these changes to the tsconfig.json
+     * Go to server/tsconfig.json and scroll to the end. Modify the "include" and "exclude" values, as specified in the tsconfig.json:
+     ```json
+      "include": ["./**/*.ts"],
+      "exclude": ["node_modules", "dist"]
+     ```
+   * The tsconfig.json file should look something like this:
+     * [`tsconfig.json`](Module13Assets/tsconfig.json) - TypeScript configuration
+
+## Step 4: Deploy the Code
+
+1. **Manual Deployment**
+   * Configure the Manual deploy option in Heroku. Everytime you push a change to your main branch, you will have to login to Heroku and click on the Deploy Branch button.
+   * In Heroku, scroll to the **Manual deploy** section.
+   * Choose the main (or preferred deployment) branch.
+   * Click **Deploy Branch** to deploy your application manually.
+
+    ![Manual Deploy](Module13Assets/ManualDeployment.png)
+
+2. **Automatic Deployment Option**
+   * Configure the Automatic deploy option in Heroku. Everytime you push a change to your main branch, Heroku will automatically publish your changes.
+   * In Heroku, scroll to the **Automatic deploys** section.
+   * Choose the main (or preferred deployment) branch.
+   * Click **Enable Automatic Deploys** to deploy your application automatically.
+
+## Step 5: Environment Variables
+
+1. **Set Up Environment Variables**
+   * In your Heroku dashboard, go to **Settings -> Config Vars**
+   * Add your MongoDB connection string as a config var:
+     ```
+     Key: MONGODB_URI
+     Value: <your MongoDB connection string>
+
+     Key: PORT
+     Value: 8000
+     ```
+   * Make sure your MongoDB connection matches the URI in your Mongo Cloud.
+   * Another option is to add the .env file to your server repository.
+
+## Step 6: Update Client Environment Variables on Render.com
+
+Update the server URL from the Render.com URL to the Heroku URL.
+
+1. **Open the Render Dashboard.**
+2. Choose the project you have created and go back to the service with type "Static Site" (your client deployment).
+3. Click on the **Environment** tab.
+4. Change the value of the `REACT_APP_SERVER_URL` environment variable to the Heroku URL (you can find this in your Heroku app's **Settings** page, under the **Domains** section).
+5. After the client finishes re-deploying, check if the client is sending queries to the new Heroku server deployment successfully.
+6. *(Optional)* If the new Heroku server deployment is working correctly, delete the server deployment from Render.
+
+## Step 7: Update Dyno Type for Memory Issues
```

Comment:
Make this a "Note" or "(Optional) Step 7", as this is an optional step.

Reply:
Done